### PR TITLE
fix(app): labware setup modal updates to match designs

### PR DESCRIPTION
**Pull request by jgbowser**
# Overview

updates the labware modal that is shown when a piece of labware is tapped on the deck map during protocol setup to match the designs

closes RAUT-530

# Test Plan

opened the deck map from the labware setup page, tapped a labware, verified the modal style matches the designs and outside click behavior works as expected

# Changelog

- switched out `LegacyModal` for `Modal` in `ProtoclSetupLabware` for the labware details modal
- made use of the `LocationIcon` component instead of the text based location display

# Review requests

ensure the PR addresses the noted issues in RAUT-530

# Risk assessment

Low

### CODE REVIEW

File: app/src/organisms/ProtocolSetupLabware/index.tsx [hunk 1]
```diff
@@ -238,15 +263,7 @@ export function ProtocolSetupLabware({
...
-                <StyledText>
-                  {mostRecentAnalysis != null
-                    ? getLabwareDisplayLocation(
-                        mostRecentAnalysis,
-                        selectedLabware.location,
-                        commandTextTranslator
-                      )
-                    : null}
-                </StyledText>
+                <Flex gridGap={SPACING.spacing4}>{location}</Flex>
```
Comment (koji) [inline app/src/organisms/ProtocolSetupLabware/index.tsx:275]: this gridGap is needed?

File: app/src/organisms/ProtocolSetupLabware/index.tsx [hunk 2]
```diff
@@ -255,7 +272,7 @@ export function ProtocolSetupLabware({
                 </StyledText>
               </Flex>
```
Comment (koji) [inline app/src/organisms/ProtocolSetupLabware/index.tsx:285]: Probably we need `nickName`
https://opentrons.slack.com/archives/CSCLVUW3C/p1686949508097869?thread_ts=1686948043.143399&cid=CSCLVUW3C

Comment (koji) [reply app/src/organisms/ProtocolSetupLabware/index.tsx:285]: Screenshot 2023-06-29 at 4 05 04 PM

File: app/src/organisms/ProtocolSetupLabware/index.tsx (hunk 1 above)
Comment (jgbowser) [reply app/src/organisms/ProtocolSetupLabware/index.tsx:275]: yeah if there are 2 icons due to it being associated with a module the grid gap provides the spacing between them

Comment (koji) [reply app/src/organisms/ProtocolSetupLabware/index.tsx:275]: oh I see, that makes sense. thank you!

**Comment by jgbowser** [comment]
@koji I have added the `nickName` to the labware details modal. Ready for a second review

**Review by koji — APPROVED** [review]
lgtm 🎉

**Comment by jgbowser** [comment]
> lgtm 🎉

@koji awesome, would you mind moving the ticket in Jira to "Done" for me? It seems mine and Eric's access to Jira and Slack has been revoked a bit too early. Much appreciated 🙏

**Comment by jgbowser** [comment]
oh... and merging this? I can't do that either now

**Comment by koji** [comment]
> oh... and merging this? I can't do that either now

😭 😭 😭 
@jgbowser 

Sure I'll update the jira ticket and merge this PR into `edge`.
Really appreciate your great work!

**Pull request merged**